test_smoke.py

```diff
@@ -61,6 +61,9 @@ def test1():
         cmake_minimum_required(VERSION 3.24)
         project(MyApp CXX)
 
+        if(${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
```

Comment:
This is a bit concerning, if we don't define this in the ``CMakeLists.txt``, would it fail?
If it doesn't fail, great, then it would be worth to add a different, new test, that defined it, and then try to check that the value is actually being used.

Reply:
In my local environment, `CMAKE_OSX_DEPLOYMENT_TARGET` is not set so `os.version` doesn't get set and everything is fine. I'm still not sure why the CI environment behaves differently, but I'll keep working on it.

After that's fixed, then yes, I plan to add test coverage for the situation that `CMAKE_OSX_DEPLOYMENT_TARGET` is set to a value in the default settings.yml file as of the minimum Conan version supported.